azure-pipelines-LGTM.yml

```diff
@@ -0,0 +1,72 @@
+# daily trigger
+trigger: none
+pr: none
+
+variables:
+- name: dotnetVersion
+  value: 5.0.100
+- name: runCodesignValidationInjection
+  value: false
+- name: NugetSecurityAnalysisWarningLevel
+  value: none
+- name: LGTM.SnapshotIdentifiers
+  value: Build=Full
+- name: LGTM.SnapshotMetadata
+  value: Owner=OPSBuild
+- name: LGTM.UploadSnapshot
+  value: true
+- name: Semmle.SkipAnalysis
+  value: false
+
+jobs:
+
+# Test and deploy on windows
+- job: OnboardLGTM
+  timeoutInMinutes: 360
+  pool:
+    name: DocFX
+  steps:
+
+  # Install .NET Core sdk
+  - task: UseDotNet@2
+    displayName: 'Install .NET Core sdk $(dotnetVersion)'
+    inputs:
+      packageType: sdk
+      version: $(dotnetVersion)
+  - task: DotNetCoreCLI@2
+    inputs:
+      command: 'restore'
+
+  - task: Semmle@0
+    displayName: Security - LGTM
+    env: 
+      SYSTEM_ACCESSTOKEN: $(MICROSOFT_PAT_TOKEN)
```

Comment:
Is there a way to get rid of this token?

Reply:
How about `$(System.AccessToken)`

Reply:
In communication with security team, Denis Levin wrote:
> LGTM Proxy does support GitHub, but since the source code repository is at GitHub and not in ADO, and ADO Pipeline’s SYSTEM_ACCESSTOKEN we receive in LGTM proxy as a parameter is created at ADO, we can’t use it to authenticate against GitHub. Instead (currently) we use this token to check if it has read access to https://dev.azure.com/microsoft account. 
> At the moment, the only fix we can offer is to set up your pipeline with another token (PAT token) scoped to https://dev.azure.com/microsoft that have read access (code_read) to dev.azure.com/Microsoft and pass this PAT token to pipeline:
SYSTEM_ACCESSTOKEN: $(MICROSOFT_PAT_TOKEN)

> We have an alternative authentication method in test environment that will take care of this issue, but it’s a few days away from deployment to production.

So we should be able to get rid of this token in a few days.